### ISSUE TITLE
SetLeaf Condition in updater_colmaker.cc

### ISSUE THREAD

**Issue by fuhaoda** [dmlc/xgboost#4504]
Based on the equation (7) in  [paper](https://arxiv.org/pdf/1603.02754.pdf)
Should the SetLeaf condition be e.best.loss_chg > params_.reg_lambda ?
```
if (e.best.loss_chg > kRtEps) {
....
  } else {
    (*p_tree)[nid].SetLeaf(e.weight * param_.learning_rate);
 }
```

**Comment by hcho3** [comment]
That’s not necessary, because we only update `e.best.loss_chg` when the loss change exceeds `min_child_weight`

**Comment by fuhaoda** [comment]
but in such case, the impact of the tuning parameter will be different, right?params_.reg_lambda only have small impact in the current implementation compared it should be in the method paper.

**Comment by hcho3** [comment]
`params_.reg_lambda` is used in the split evaluator class to compute the loss change:
https://github.com/dmlc/xgboost/blob/17df5fd296aaff18123c53a777f24111b3dd7336/src/tree/split_evaluator.cc#L135-L158
